modules/apps/29-fee/ibc_module_test.go

```diff
@@ -468,3 +470,301 @@ func (suite *FeeTestSuite) TestOnChanCloseConfirm() {
 		})
 	}
 }
+
+func (suite *FeeTestSuite) TestOnRecvPacket() {
+	testCases := []struct {
+		name          string
+		malleate      func(suite *FeeTestSuite)
+		sourceRelayer bool
+	}{
+		{
+			"success",
+			func(suite *FeeTestSuite) {
+				suite.chainA.GetSimApp().IBCFeeKeeper.SetCounterpartyAddress(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String())
```

Comment:
I'm not actually sure what the idiomatic way is (maybe @colin-axner @damiannolan disagrees with me here) but I'd prefer to have the success case do nothing and put this `SetCounterpartyAddress` below instead of calling here. Instead, having the malleate change the relayer address to an empty string explicitly for the fail case would be easier to read IMO.

Just a nit, maybe this is totally fine :)

Reply:
Actually, I think it makes more sense the way you have it after looking at it closer.

Reply:
I agree that success case should do nothing. This makes it easier to verify the failing tests cases (otherwise it can be easy to forget to do the required change)